_rules/active-focus-shortcut-1e9941.md

```diff
@@ -0,0 +1,255 @@
+---
+id: 1e9941
+name: Printable key shortcut is only active when component has focus
+rule_type: atomic
+description: |
+  This rule checks that if keyboard shortcuts for a user interface component are implemented using only printable characters, then they are only available when that component has focus.
+accessibility_requirements: # Remove whatever is not applicable
+  wcag21:2.1.4: # Character Key Shortcuts (A)
+    forConformance: true
+    failed: not satisfied
+    passed: satisfied
+    inapplicable: further testing needed
+input_aspects:
+  - DOM Tree
+  - CSS Styling
+acknowledgements:
+  authors:
+    - João Vicente
+    - Carlos Duarte
+---
+
+## Applicability
+
+The rule applies to any [HTML document][] with at least one [keyboard shortcut][] that requires pressing only [printable character][] keys to trigger an action on a [user interface component][].
+
+## Expectation
+
+For each [user interface component][] that is a [descendant][] of the root node of the test target, the [printable character][] [shortcut][keyboard shortcut] that triggers the action of that [user interface component][] can only be triggered when that component has [focus][].
+
+## Assumptions
+
+This rule assumes as applicable [keyboard shortcuts][keyboard shortcut] those implemented by the test target [content][]. Any other means (e.g. browser extensions, browser settings, user agents, external browser applications) are not considered.
+
+## Accessibility Support
+
+_There are no major accessibility support issues known for this rule._
+
+## Background
+
+- [Understanding Success Criterion 2.1.4: Character Key Shortcuts](https://www.w3.org/WAI/WCAG21/Understanding/character-key-shortcuts.html)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+The [HTML document][] has a single [printable character][] [keyboard shortcut][] for a [user interface component][], which is only available when that component has focus.
+
+```html
+<html>
+  <head>
+    <title>Passed Example 1 for rule 1e9941</title>
+    <script src="/test-assets/ffbc54/shortcut.js"></script>
+  </head>
+
+  <body onload="shortcut({focusOnly: true})">
+    <label for="text">Add to list (press + to add):</label>
+    <input type="text" id="target" />
+    <br />
+    <div>
+      To do list
+    </div>
+    <ul id="list"></ul>
+  </body>
+</html>
+```
+
+#### Passed Example 2
+
+The [HTML document][] has a single [printable character][] [keyboard shortcut][] for a [user interface component][], which is only available when that component has focus, and another [keyboard shortcut][] that requires pressing one [non-printable character][non-printable characters].
+
+```html
+<html>
+  <head>
+    <title>Passed Example 2 for rule 1e9941</title>
+    <script src="/test-assets/ffbc54/shortcut.js"></script>
+  </head>
+
+  <body onload="shortcut({focusOnly: true}, {shortcutKey: '«' , ctrlKey: true})">
+    <label for="text">Add to list (press + or ctrl+« to add):</label>

```

Comment:
UI think that `«` is not a very good key to use. I am not sure it even exists on US keyboards… Maybe simply use `a` (as it "a"dds an item).